### ISSUE TITLE
new rule - ban nested if expressions

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#13564]
i keep stumbling across code where, for whatever reason, the person writing it thought it was a good idea to have several nested conditions in an `if` expression, which looks absolutely awful:

```py
foo = (
    "a"
    if bool()
    else str(
        "b"
        if bool()
        else (
            "c"
            if bool()
            else str(
                "d"
                if bool()
                else "e"
            )
        )
    )
)
```

it would be nice if ruff had a rule to enforce that if expressions do not have any nested if expressions inside them, and instead suggests writing it as a regular if statement which is far easier to read

**Comment by lsouder-ozone3d** [comment]
It's possible you just want some combination of:
- [too-many-nested-blocks (PLR1702)](https://docs.astral.sh/ruff/rules/too-many-nested-blocks/#too-many-nested-blocks-plr1702)
- [complex-structure (C901)](https://docs.astral.sh/ruff/rules/complex-structure/#complex-structure-c901)

**Comment by DetachHead** [comment]
hmm, it looks like neither of those rules report on if expressions no matter how nested i make it
